src/main.cpp

```diff
@@ -743,22 +743,17 @@ int main(int argc, char**argv)
 
   DiameterResolver* diameter_resolver = NULL;
   RealmManager* realm_manager = NULL;
-  Diameter::Peer* peer = NULL;
-
-  if (!options.dest_realm.empty())
+  
+  if (hss_configured)
   {
     diameter_resolver = new DiameterResolver(dns_resolver, af);
     realm_manager = new RealmManager(diameter_stack,
                                      options.dest_realm,
+                                     options.dest_host,
                                      options.max_peers,
                                      diameter_resolver);
     realm_manager->start();
   }
-  else if (!(options.dest_host.empty() || options.dest_host == "0.0.0.0"))
```

Comment:
No, we're safe - see https://github.com/Metaswitch/homestead/blob/realmmanager_with_host/src/main.cpp#L689, where hss_configured is false of options.dest_host == "0.0.0.0".